doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -0,0 +1,874 @@
+=================================
+Term rewriting and simplification
+=================================
```

Comment:
Yes, we should at least link between the two chapters or even possibly combine them (even if "Generalized Rewriting" is a programmable mechanism, which is why it is where it is as of today).

As for the file name, I prefer to keep them as short as possible since they matter for URLs.